pkg/agent/controller/egress/egress_controller.go

```diff
@@ -569,20 +570,54 @@ func (c *EgressController) updateEgressStatus(egress *crdv1a2.Egress, egressIP s
 	toUpdate := egress.DeepCopy()
 	var updateErr, getErr error
 	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		var status v1.ConditionStatus
+		var statusReason, statusMessage string
 		if isLocal {
 			// Do nothing if the current EgressNode in status is already this Node.
 			if toUpdate.Status.EgressNode == c.nodeName && toUpdate.Status.EgressIP == egressIP {
 				return nil
 			}
 			toUpdate.Status.EgressNode = c.nodeName
 			toUpdate.Status.EgressIP = egressIP
+			status = v1.ConditionTrue
+			statusReason = "Assigned"
+			statusMessage = "EgressIP and EgressNode successfully assigned"
 		} else {
-			// Do nothing if the current EgressNode in status is not this Node.
-			if toUpdate.Status.EgressNode != c.nodeName {
+			// If the Egress is not schedulable at all, do nothing.
+			if !isEgressSchedulable(toUpdate) {
+				return nil
+			}
+			// Select one Node to update false status among all Nodes.
+			nodeToUpdateStatus, err := c.cluster.SelectNodeForIP(egress.Spec.EgressIP, "")
+			if err != nil {
+				return err
+			}
+			// Skip if the Node is not the selected one.
+			if nodeToUpdateStatus != c.nodeName {
 				return nil
 			}
 			toUpdate.Status.EgressNode = ""
 			toUpdate.Status.EgressIP = ""
+			status = v1.ConditionFalse
+			statusReason = "EgressNode Error"
+			statusMessage = "No available Node to be elected as Egress Node"
```

Comment:
Maybe name the reason `NoAvailableNode`.
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties
> In condition types, and everywhere else they appear in the API, Reason is intended to be a one-word, CamelCase representation of the category of cause of the current status, and Message is intended to be a human-readable phrase or sentence, which may contain specific details of the individual occurrence. Reason is intended to be used in concise output, such as one-line kubectl get output, and in summarizing occurrences of causes, whereas Message is intended to be presented to users in detailed status explanations, such as kubectl describe output.
And some example of existing conditions:
```
    conditions:
    - lastTransitionTime: "2023-07-18T16:37:48Z"
      lastUpdateTime: "2023-07-18T16:37:48Z"
      message: Deployment has minimum availability.
      reason: MinimumReplicasAvailable
      status: "True"
      type: Available
    - lastTransitionTime: "2023-07-18T16:37:18Z"
      lastUpdateTime: "2023-07-18T16:37:48Z"
      message: ReplicaSet "server-7bc7f69979" has successfully progressed.
      reason: NewReplicaSetAvailable
      status: "True"
      type: Progressing
...
    conditions:
    - lastHeartbeatTime: "2023-07-25T08:26:34Z"
      lastTransitionTime: "2023-07-18T16:34:52Z"
      message: kubelet has sufficient memory available
      reason: KubeletHasSufficientMemory
      status: "False"
      type: MemoryPressure
    - lastHeartbeatTime: "2023-07-25T08:26:34Z"
      lastTransitionTime: "2023-07-18T16:34:52Z"
      message: kubelet has no disk pressure
      reason: KubeletHasNoDiskPressure
      status: "False"
      type: DiskPressure
    - lastHeartbeatTime: "2023-07-25T08:26:34Z"
      lastTransitionTime: "2023-07-18T16:34:52Z"
      message: kubelet has sufficient PID available
      reason: KubeletHasSufficientPID
      status: "False"
      type: PIDPressure
    - lastHeartbeatTime: "2023-07-25T08:26:34Z"
      lastTransitionTime: "2023-07-18T16:37:34Z"
      message: kubelet is posting ready status
      reason: KubeletReady
      status: "True"
      type: Ready
```